src/typeahead/typeahead-window.ts

```diff
@@ -56,13 +56,13 @@ export class NgbTypeaheadWindow {
    */
   @Output('select') selectEvent = new EventEmitter();
 
-  getActive() { return this.results[this._activeIdx]; }
+  getActive() { return this.results[this.activeIdx]; }
 
-  markActive(_activeIdx: number) { this._activeIdx = _activeIdx; }
+  markActive(activeIdx: number) { this.activeIdx = activeIdx; }
```

Comment:
Yes, somewhat, thanks. 

My main doubt is with `next()`, `prev()` and `getActive()` in typeahead-window. 

As you said, this component is not exported from the module, so it doesn't matter much if these methods are internal or not.

These methods, which are not called from the template, exist and are public because typeahead needs to call them. Not because they're meant to be part of the public external API of the component (at least not now). So I would tend to make the internal, too, even though they're not called from the template. 

And in fact, the same goes with all the methods implementing Angular interfaces: they're not called from the template either, and are not supposed to be called by anything other than the framework, so these should be internal, too, in my mind.

Anyway, I made `markActive()` and `select()` internal, now, so these changes should follow the rules.

Tell me if anything else must be changed.